### PR TITLE
Mobile menu

### PR DESCRIPTION
**Pull request by wackerow**
### Description
Adds new component `MobileMenu.tsx` according to design utilizing Chakra-UI Modal components.

### CODE REVIEW

**Review by corwintines — APPROVED** [review]
LGTM, I pushed a couple commits to abstract some duplicate code I saw, and setup for search, but nice one!

**Comment by wackerow** [comment]
@corwintines 
image
image

Getting broken styling in the nav bar now. Just noting the styling for those buttons is different for the nav vs the mobile menu that I added, that was why I kept them separated.

**Comment by corwintines** [comment]
@wackerow pushed up a fix for that

**Comment by wackerow** [comment]
Cool looks good, would approve it but it was my PR

File: src/components/UI/HeaderButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  }
+  return (
+    <Flex direction={{base: 'column', md: 'row'}}>
+     {/* DOWNLOADS */}
+     <Stack {...menuItemStyles}>
+        <NextLink href={DOWNLOADS_PAGE} passHref>
+          <Link _hover={{ textDecoration: 'none' }} onClick={close}>
+            <Text textStyle={{ base: 'header-mobile-button', md: 'header-button' }}>
+              downloads
```
Comment (nhsz) [inline src/components/UI/HeaderButtons.tsx:34]: text [should be centered](https://www.figma.com/file/ekzIgwyeVKLtFSAcnA0Q0D/geth-website?node-id=885%3A8258&t=Ic6WpLw62Cylvxti-0)

File: src/components/UI/HeaderButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          </Link>
+        </NextLink>
+      </Stack>
+
+      {/* DOCUMENTATION */}
+      <Stack {...menuItemStyles}>
+        <NextLink href={DOCS_PAGE} passHref>
+          <Link _hover={{ textDecoration: 'none' }} onClick={close}>
+            <Text textStyle={{ base: 'header-mobile-button', md: 'header-button' }}>
+              documentation
```
Comment (nhsz) [inline src/components/UI/HeaderButtons.tsx:45]: text [should be centered](https://www.figma.com/file/ekzIgwyeVKLtFSAcnA0Q0D/geth-website?node-id=885%3A8258&t=Ic6WpLw62Cylvxti-0)

File: src/components/UI/search/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export const Search: FC = () => {
+  return (
+    <Stack
+      borderBottom={{ base: BORDER_WIDTH, md: 'none' }}
+      borderRight={{ base: 'none', md: BORDER_WIDTH }}
+      borderColor={{ base: 'bg', md: 'primary' }}
+      p={{ base: 8, md: 4 }}
+      _hover={{ base: {bg: 'primary'}, md: {bg: 'none'}}}
+    >
+      <InputGroup>
```
Comment (nhsz) [inline src/components/UI/search/Search.tsx:16]: we use [less padding for search input](text [should be centered](https://www.figma.com/file/ekzIgwyeVKLtFSAcnA0Q0D/geth-website?node-id=885%3A8258&t=Ic6WpLw62Cylvxti-0))

File: src/components/layouts/MobileMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        <HamburgerIcon />
+      </Box>
+
+      {/* MODAL */}
+      <Modal isOpen={isOpen} onClose={onClose} motionPreset='none'>
+
+        <ModalOverlay />
+        <ModalContent>
+          {/* MOBILE MENU */}
+          <Flex
```
Comment (nhsz) [inline src/components/layouts/MobileMenu.tsx:40]: container reaches the botto of the screen and [should have bottom padding too](https://www.figma.com/file/ekzIgwyeVKLtFSAcnA0Q0D/geth-website?node-id=885%3A8258&t=Ic6WpLw62Cylvxti-0)

Screen Shot 2022-11-23 at 12 37 52

**Review by nhsz — CHANGES_REQUESTED** [review]
Nice! Thanks @wackerow. Some adjustments needed to match the design on figma.

**Pull request merged**